autogpt/llm/base.py

```diff
@@ -73,17 +74,37 @@ class ChatSequence:
     """Utility container for a chat sequence"""
 
     model: ChatModelInfo
-    messages: list[Message] = field(default_factory=list)
+    messages: list[Message] = field(default_factory=list[Message])
 
-    def __getitem__(self, i: int):
-        return self.messages[i]
+    @overload
+    def __getitem__(self, key: int) -> Message:
+        ...
+
+    @overload
+    def __getitem__(self, key: slice) -> ChatSequence:
+        ...
+
+    def __getitem__(self, key: int | slice):
+        if isinstance(key, slice):
+            copy = deepcopy(self)
+            copy.messages = self.messages[key]
+            return copy
+        return self.messages[key]
 
     def __iter__(self):
         return iter(self.messages)
 
     def __len__(self):
         return len(self.messages)
 
+    def add(
+        self,
+        message_role: MessageRole,
+        content: str,
+        type: MessageType | None = None,
+    ):
+        self.messages.append(Message(message_role, content, type))
```

Comment:
Use the methods present on the class.
```suggestion
        self.append(Message(message_role, content, type))
```